sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -0,0 +1,305 @@
+
+    def _get_input_artifact_type(
+        input_name: str,
+        inputs: List[structures.InputSpec],
+    ) -> str:
+      """Find the input artifact type by input name.
+
+      Args:
+        input_name: The name of the component input.
+        inputs: The list of InputSpec
+
+      Returns:
+        The input type name if found in inputs, or '' if not found.
+      """
+      for component_input in inputs:
+        if component_input.name == input_name:
+          return type_utils.get_artifact_type(component_input.type)
+      return ''
+
+    deployment_config = pipeline_spec_pb2.PipelineDeploymentConfig()
+    importer_tasks = []
+
+    for op in pipeline.ops.values():
+      component_spec = op._metadata
+      task = pipeline_spec.tasks.add()
+      task.CopyFrom(op.task_spec)
+      deployment_config.executors[task.executor_label].container.CopyFrom(
+          op.container_spec)
+
+      # Check if need to insert importer node
+      for input_name in task.inputs.artifacts:
+        if not task.inputs.artifacts[input_name].producer_task:
+          artifact_type = _get_input_artifact_type(input_name,
+                                                   component_spec.inputs)
+
+          importer_task, importer_spec = importer_node.build_importer_spec(
+              task, input_name, artifact_type)
+          importer_tasks.append(importer_task)
+
+          task.inputs.artifacts[
+              input_name].producer_task = importer_task.task_info.name
+          task.inputs.artifacts[
+              input_name].output_artifact_key = importer_node.OUTPUT_KEY
+
+          deployment_config.executors[
+              importer_task.executor_label].importer.CopyFrom(importer_spec)
+
+    pipeline_spec.deployment_config.Pack(deployment_config)
+    pipeline_spec.tasks.extend(importer_tasks)
+
+    return pipeline_spec
+
+  def _create_pipeline(
+      self,
+      pipeline_func: Callable[..., Any],
+      pipeline_name: Optional[str] = None,
+      pipeline_description: Optional[str] = None,
+      params_list: List[dsl.PipelineParam] = None,
+  ) -> pipeline_spec_pb2.PipelineSpec:
+    """Internal implementation of create_pipeline.
+
+    Args:
+      pipeline_func: Pipeline function with @dsl.pipeline decorator.
+      pipeline_name: The name of the pipeline. Optional.
+      pipeline_description: An optional description of the pipeline.
+      params_list: A list of pipeline arguments.
+
+    Returns:
+      The IR representation (pipeline spec) of the pipeline.
+    """
+    params_list = params_list or []
+
+    # Create the arg list with no default values and call pipeline function.
+    # Assign type information to the PipelineParam
+    pipeline_meta = _python_op._extract_component_interface(pipeline_func)
+    pipeline_meta.name = pipeline_name or pipeline_meta.name
+    pipeline_meta.description = pipeline_description or pipeline_meta.description
+    pipeline_name = sanitize_k8s_name(pipeline_meta.name)
```

Comment:
I'm wondering if this is needed. This maps to a context name. Can we drop this for now, and add it when we need it?